param/parameterized.py

```diff
@@ -1600,66 +1685,120 @@ def __setstate__(self,state):
             setattr(self,name,value)
         self.initialized=True
 
-
-    @classmethod
-    def params(cls,parameter_name=None):
+    def __repr__(self):
         """
-        Return the Parameters of this class as the
-        dictionary {name: parameter_object}
+        Provide a nearly valid Python representation that could be used to recreate
+        the item with its parameters, if executed in the appropriate environment.
 
-        Includes Parameters from this class and its
-        superclasses.
+        Returns 'classname(parameter1=x,parameter2=y,...)', listing
+        all the parameters of this object.
         """
-        # CB: we cache the parameters because this method is called often,
-        # and parameters are rarely added (and cannot be deleted)
-        try:
-            pdict=getattr(cls,'_%s__params'%cls.__name__)
-        except AttributeError:
-            paramdict = {}
-            for class_ in classlist(cls):
-                for name,val in class_.__dict__.items():
-                    if isinstance(val,Parameter):
-                        paramdict[name] = val
+        settings = ['%s=%s' % (name,repr(val))
+                    for name,val in self.get_param_values()]
```

Comment:
I'm struggling with the diff here, so maybe I'm confused...but won't this kind of thing trigger the deprecation warning?